### PR TITLE
Treat standalone lesson and lesson without published course the same

### LINKED ISSUE #2681: Feedback - Exit Course link on lesson leads to 404 when parent course is unpublished
## Type of feedback
//dev

## Description
"Exit Course" link on main Lessons pages sometimes works and sometimes 404s. I didn't click through all links, but it looks like "Exit Course" lessons for the Beginner WordPress Developer work and "Exit Course" lessons for Advanced WordPress Developer do not.

Examples where the Exit Course link doesn't work:

- https://learn.wordpress.org/lesson/adding-custom-user-roles-using-a-wordpress-plugin/
- https://learn.wordpress.org/lesson/using-caching-to-improve-website-performance-w3-total-cache-plugin/
- https://learn.wordpress.org/lesson/wordpress-taxonomies-2/
- https://learn.wordpress.org/lesson/creating-custom-post-types-and-taxonomies/

Examples where it does:
- https://learn.wordpress.org/lesson/website-optimization/
- https://learn.wordpress.org/lesson/wordpress-hooks/
- https://learn.wordpress.org/lesson/time-to-test-your-skills/

## Step-by-step reproduction instructions (optional)
1. From learn.wordpress.org, click on [Lessons](https://learn.wordpress.org/lessons/) in the Learn navigation (logged in state doesn't matter)
2. Click on the [Adding custom user roles using a WordPress plugin](https://learn.wordpress.org/lesson/adding-custom-user-roles-using-a-wordpress-plugin/
) lesson
3. From top right screen (at least that's the position on desktop), click the [Exit Course](https://learn.wordpress.org/?post_type=course&p=257841) Link
4. You're redirected to a "Sorry, we couldn't find this page" page. Note redirect URL structure: https://learn.wordpress.org/?post_type=course&p=257841

## Screenshots or screen recording (optional)
Screenshot 2024-07-16 at 8 14 17 AM

## Other information (optional)
Testing on latest MacOS + Chrome

### PR DESCRIPTION
**Pull request by adamwoodnz**
Fixes #2681 

Alternative to https://github.com/WordPress/Learn/pull/2740

The idea here is to fix issues with parts of the Lesson UI that expect a published parent course, by treating lessons like this the same as we do standalone lessons. Content creators can compose the course with modules and lessons, and publish the lessons before the course is published, without things like the 'Exit course' link and course outline showing in the frontend, and resulting in 404s if the parent course is accessed.

| Standard Lesson belonging to Course and Module | Lesson in unpublished course |
|--------|--------|
| localhost_8888_lesson_wordpress-and-web-servers_(Desktop) (2) | localhost_8888_lesson_lesson-in-unpublished-course_(Desktop) | 

### Testing
1. Regression test a standard lesson with a parent course. Whether or not a module is used shouldn't matter.
2. Regression test a standalone lesson (lesson with no assigned course). The only difference with prod should be the addition of the `Exit lesson` link in the top left, which goes to the lessons archive.
3. Create a course with a lesson. Publish the lesson but not the course.
4. View the lesson on the frontend, it should display as if standalone.

### CODE REVIEW

File: wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	<!-- wp:group {"style":{"spacing":{"padding":{"left":"var:preset|spacing|edge-space","right":"var:preset|spacing|edge-space","top":"0px","bottom":"0px"}}},"backgroundColor":"white","className":"sensei-course-theme-header-content","layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
+	<div class="wp-block-group sensei-course-theme-header-content has-white-background-color has-background" style="padding-top:0px;padding-right:var(--wp--preset--spacing--edge-space);padding-bottom:0px;padding-left:var(--wp--preset--spacing--edge-space)">
+
+		<!-- wp:wporg/site-breadcrumbs {"fontSize":"small","style":{"spacing":{"padding":{"top":"18px","bottom":"18px"}}}} /-->
+
+		<!-- wp:group {"style":{"spacing":{"blockGap":"12px"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+		<div class="wp-block-group">
+
+			<!-- wp:paragraph {"className":"is-style-short-text","fontSize":"small"} -->
+			<p class="is-style-short-text has-small-font-size"><a href="<?php echo esc_url( get_post_type_archive_link( 'lesson' ) ); ?>">Exit lesson</a></p>
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php:22]: 'Exit course' link, which redirects to the parent course on normal lessons, replaced with a link to the Lessons archive

File: wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	<!-- wp:group {"style":{"spacing":{"padding":{"left":"var:preset|spacing|edge-space","right":"var:preset|spacing|edge-space","top":"0px","bottom":"0px"}}},"backgroundColor":"white","className":"sensei-course-theme-header-content","layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
+	<div class="wp-block-group sensei-course-theme-header-content has-white-background-color has-background" style="padding-top:0px;padding-right:var(--wp--preset--spacing--edge-space);padding-bottom:0px;padding-left:var(--wp--preset--spacing--edge-space)">
+
+		<!-- wp:wporg/site-breadcrumbs {"fontSize":"small","style":{"spacing":{"padding":{"top":"18px","bottom":"18px"}}}} /-->
+
+		<!-- wp:group {"style":{"spacing":{"blockGap":"12px"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+		<div class="wp-block-group">
+
+			<!-- wp:paragraph {"className":"is-style-short-text","fontSize":"small"} -->
+			<p class="is-style-short-text has-small-font-size"><a href="<?php echo esc_url( get_post_type_archive_link( 'lesson' ) ); ?>"><?php echo esc_html_e( 'Exit lesson', 'wporg-learn' ); ?></a></p>
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php:22]: 'Exit course' link, which redirects to the parent course on normal lessons, replaced with a link to the Lessons archive

File: wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-columns.php
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-columns.php:1]: Extracted from single-lesson.html for conditional rendering

File: wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-standalone.php:1]: This is mostly the same as `sensei-lesson-header.php` + `sensei-lesson-columns.php`, but with the course outline, course header content (progress and exit course link), and notices removed.

File: wp-content/themes/pub/wporg-learn-2024/src/style/_sensei.scss [hunk 3]
```diff
@@ -271,32 +275,11 @@ body.sensei {
...
-
-			.sensei-course-theme__main-content {
-				margin-left: auto !important;
-				margin-right: auto !important;
-				max-width: var(--wp--style--global--content-size);
-				padding-left: var(--wp--preset--spacing--edge-space);
-				padding-right: var(--wp--preset--spacing--edge-space);
-			}
-		}
-
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/style/_sensei.scss:299]: Don't need to do a lot of this CSS hacking now we have a dedicated pattern for standalone.

**Review by outdoor2kode — APPROVED** [review]
LGTM 👍 

- [x] Regression test a standard lesson with a parent course. Whether or not a module is used shouldn't matter.
- [x] Regression test a standalone lesson (lesson with no assigned course). The only difference with prod should be the addition of the Exit lesson link in the top left, which goes to the lessons archive.
- [x] Create a course with a lesson. Publish the lesson but not the course.
- [x] View the lesson on the frontend, it should display as if standalone.
- [x] Publish the course, the lesson ^ displays as belonging to the course.

**Pull request merged**